Arm64/aarch64 build failed

**Describe the bug**
Arm64/aarch64 build fails

**To Reproduce**
Steps to reproduce the behavior:
1. `mayastor-source-code-dir]$ ./build.sh`
```
...
Compiler for C supports arguments -Wmissing-prototypes: YES
Compiler for C supports arguments -Wnested-externs: YES
Compiler for C supports arguments -Wold-style-definition: YES
Compiler for C supports arguments -Wpointer-arith: YES
Compiler for C supports arguments -Wsign-compare: YES
Compiler for C supports arguments -Wstrict-prototypes: YES
Compiler for C supports arguments -Wundef: YES
Compiler for C supports arguments -Wwrite-strings: YES
Compiler for C supports arguments -Wno-address-of-packed-member: YES
Compiler for C supports arguments -Wno-packed-not-aligned: YES
Compiler for C supports arguments -Wno-missing-field-initializers: YES
Program /build/source/dpdk/config/arm/armv8_machine.py found: YES (/build/source/dpdk/config/arm/armv8_machine.py)

config/arm/meson.build:474:16: ERROR: Problem encountered: Error when getting Arm Implementer ID and part number.

A full log can be found at /build/source/dpdk/build-tmp/meson-logs/meson-log.txt
make[1]: *** [Makefile:195: /build/source/dpdk/build-tmp] Error 1
make: *** [/build/source/mk/spdk.subdirs.mk:44: dpdkbuild] Error 2

```

```
$ cat build.sh
#!/bin/bash -e

images="mayastor-io-engine"
tag="develop"
docker_hub_user="USER"
cut_str=""


if [[ -n $1 ]];then
        type=$1
else
        type="release"
fi

if [[ "$type" == "debug" ]]; then
    tag=${tag}-debug
        images="mayastor-io-engine-dev"
        cut_str="-dev"
fi

# build and tag images
for image in $images; do
    file="artifacts/docker/${image}.tar.gz"

    echo "Building ${image} to $file"
    nix --experimental-features 'nix-command flakes' build -v -f . -o $file \
                images.${image}

    echo "Loading ${file}..."
    docker load --quiet --input ${file} \
        | awk '{ print $3 }' \
        | ( \
        read img; \
        LATEST=$(echo ${img} | awk '{split($0,a,":"); print a[1]}'); \
        LATEST=${LATEST%${cut_str}}:$tag; \
        LATEST=${docker_hub_user}/$(echo ${LATEST} | awk '{split($0,a,"/"); print a[2]}') ; \
        echo "Tagging ${img} to ${LATEST} (from ${file})."; \
        docker tag ${img} ${LATEST}; \
        docker push ${LATEST}; \
        docker image rm ${img}; \
        );
done
docker image prune -f
docker images

```

**Expected behavior**
Build success.

**Screenshots**
If applicable, add screenshots to help explain your problem.

** OS info (please complete the following information):**
 - Distro: ubuntu 20.04
 - Kernel version:  v5.4
 - MayaStor develop branch

**Additional context**
Add any other context about the problem here.

--- a/nix/pkgs/libspdk/default.nix
+++ b/nix/pkgs/libspdk/default.nix
@@ -104,9 +104,16 @@ let
 
     hardeningDisable = [ "all" ];
 
-    buildPhase = ''
-      make -j`nproc`
-    '';
+    buildPhase = (if (targetPlatform.config == "aarch64-unknown-linux-gnu") then
+      [
+        "DPDKBUILD_FLAGS=-Dplatform=generic"
+      ]
+    else
+      [ ]
+    ) ++
+    [
+      "make -j`nproc`"
+    ];
 
     installPhase = ''
       echo "installing SPDK to $out"

